target/linux/ath79/dts/ar9331_pisen_ts-d084.dts

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9331.dtsi"
+
+/ {
+	model = "Pisen TS-D084";
+	compatible = "pisen,ts-d084", "qca,ar9331";
+
+	aliases {
+		serial0 = &uart;
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
```

Comment:
"gpio-keys" should work and it has less overhead.

(Note: if it works as it is supposed to, you could do away with the poll-interval property.)